core/src/main/scala/org/apache/spark/scheduler/BlacklistTracker.scala

```diff
@@ -0,0 +1,130 @@
+        conf.get(config.BLACKLIST_LEGACY_TIMEOUT_CONF) match {
+          case Some(legacyTimeout) =>
+            if (legacyTimeout == 0) {
+              logWarning(s"Turning off blacklisting due to legacy configuaration:" +
+                s" $legacyKey == 0")
+              false
+            } else {
+              // mostly this is necessary just for tests, since real users that want the blacklist
+              // will get it anyway by default
+              logWarning(s"Turning on blacklisting due to legacy configuration:" +
+                s" $legacyKey > 0")
+              true
+            }
+          case None =>
+            // local-cluster is *not* considered local for these purposes, we still want the
+            // blacklist enabled by default
+            !Utils.isLocalMaster(conf)
+        }
+    }
+  }
+
+  def getBlacklistTimeout(conf: SparkConf): Long = {
+    conf.get(config.BLACKLIST_TIMEOUT_CONF).getOrElse {
+      conf.get(config.BLACKLIST_LEGACY_TIMEOUT_CONF).getOrElse {
+        Utils.timeStringAsMs(DEFAULT_TIMEOUT)
+      }
+    }
+  }
+
+  /**
+   * Verify that blacklist configurations are consistent; if not, throw an exception.  Should only
+   * be called if blacklisting is enabled.
+   *
+   * The configuration for the blacklist is expected to adhere to a few invariants.  Default
+   * values follow these rules of course, but users may unwittingly change one configuration
+   * without making the corresponding adjustment elsewhere.  This ensures we fail-fast when
+   * there are such misconfigurations.
+   */
+  def validateBlacklistConfs(conf: SparkConf): Unit = {
+
+    def mustBePos(k: String, v: String): Unit = {
+      throw new IllegalArgumentException(s"$k was $v, but must be > 0.")
+    }
+
+    // undocumented escape hatch for validation -- just for tests that want to run in an "unsafe"
+    // configuration.
+    if (!conf.get("spark.blacklist.testing.skipValidation", "false").toBoolean) {
+
+      Seq(
+        config.MAX_TASK_ATTEMPTS_PER_EXECUTOR,
+        config.MAX_TASK_ATTEMPTS_PER_NODE,
+        config.MAX_FAILURES_PER_EXEC_STAGE,
+        config.MAX_FAILED_EXEC_PER_NODE_STAGE
+      ).foreach { config =>
+        val v = conf.get(config)
+        if (v <= 0) {
+          mustBePos(config.key, v.toString)
+        }
+      }
+
+      val timeout = getBlacklistTimeout(conf)
+      if (timeout <= 0) {
+        // first, figure out where the timeout came from, to include the right conf in the message.
+        conf.get(config.BLACKLIST_TIMEOUT_CONF) match {
+          case Some(t) =>
+            mustBePos(config.BLACKLIST_TIMEOUT_CONF.key, timeout.toString)
+          case None =>
+            mustBePos(config.BLACKLIST_LEGACY_TIMEOUT_CONF.key, timeout.toString)
+        }
+      }
+
+      val maxTaskFailures = conf.getInt("spark.task.maxFailures", 4)
+      val maxNodeAttempts = conf.get(config.MAX_TASK_ATTEMPTS_PER_NODE)
+
+      if (maxNodeAttempts >= maxTaskFailures) {
```

Comment:
its not really equivalent to disabling blacklisting -- if we allowed it, you'd still get executor level blacklisting.  It would just be a lack of effective node blacklisting, which means that if you had multiple executors on a bad node you wouldn't blacklist effectively.

I know I am quibbling but I also worry that users will not understand the problem -- they'll think they've enabled executor level blacklisting, so the failure won't make sense.  I thought the original "one bad node" was actually clearer.  Would it help to add another sentence explaining the multiple executors?

"Though blacklisting is enabled, with this configuration, Spark will not be robust to one bad node.  Since one node may have multiple executors, executor blacklisting is insufficient on its own."

honestly dont' know if I feel any better about that, just hoping it may get you to think of something else ...